src/progressive-profiling/ProgressiveProfiling.jsx

```diff
@@ -54,13 +58,23 @@ const ProgressiveProfiling = (props) => {
       })
       .catch(() => {});
 
-    if (props.location.state && props.location.state.registrationResult) {
-      setRegistrationResult(props.location.state.registrationResult);
+    let userEnrollmentAction = false;
+    if (registrationResponse) {
+      setRegistrationResult(registrationResponse);
       sendPageEvent('login_and_registration', 'welcome');
+
+      const queryParams = getAllPossibleQueryParams(registrationResponse.redirectUrl);
+      if ('enrollment_action' in queryParams) {
+        userEnrollmentAction = true;
+      }
+    }
+
+    if (enablePersonalizedRecommendations && !userEnrollmentAction) {
+      setShowRecommendationsPage(true);
     }
-  }, [DASHBOARD_URL, props.location.state]);
+  }, [DASHBOARD_URL, enablePersonalizedRecommendations, location.state, registrationResponse]);
```

Comment:
no need for location.state as a dependency. 
```suggestion
  }, [DASHBOARD_URL, enablePersonalizedRecommendations, registrationResponse]);
```